src/Engine/GameAction.java

```diff
@@ -93,6 +118,8 @@ public boolean isReadyToExecute()
    * Performs the built-up action, using the passed-in GameMap.
    * @return true if the action successfully executes, false if a problem occurs.
    */
+  // TODO: Should this validate that the GameAction is valid (e.g. we aren't trying
+  //  to move an Infantry unit 10 spaces)?
```

Comment:
Well, it's an interesting problem. We don't want to show the user invalid actions, so we kinda need to validate the user actions at each step (and then in theory it won't be possible to construct a GameAction that isn't correct). But then an AI doesn't go through the same process. If you manually build `GameAction(Infantry, 1, 1, WAIT, 10, 5)` and call execute, it will happily move the infantry 13 spaces regardless of the destination tile type. I'm somewhat inclined to add proper validation of some kind here as well, just as a final fail-safe (and for the extra debugging info it would provide when running test cases).